src/qml/MapCanvas.qml

```diff
@@ -104,6 +104,116 @@ Item {
+    const currentRotation = mapCanvasWrapper.mapSettings.rotation;
+    jumpDetails.fromScale = mapCanvasWrapper.mapSettings.scale;
+    jumpDetails.fromRotation = currentRotation;
+    jumpDetails.fromX = currentCenter.x;
+    jumpDetails.fromY = currentCenter.y;
+    jumpDetails.toX = point.x;
+    jumpDetails.toY = point.y;
+    jumpDetails.toScale = scale;
+    jumpDetails.completedCallback = callback;
+    if (rotation !== -1) {
+      const rotationDiff = shortestRotationPath(currentRotation, rotation);
+      jumpDetails.rotationDelta = rotationDiff;
+      jumpDetails.toRotation = rotation;
+    } else {
+      jumpDetails.toRotation = -1;
+    }
+    jumpDetails.position = 0.0;
+    jumpDetails.handleMargins = handleMargins;
+    freeze('jumping');
+    jumpDetails.enabled = true;
+    jumpDetails.position = 1.0;
+  }
+
+  Item {
+    id: jumpDetails
+
+    property bool enabled: false
+    property double fromRotation: 0
+    property double fromScale: 0
+    property double fromX: 0
+    property double fromY: 0
+    property double toRotation: -1
+    property double toScale: -1
+    property double toX: 0
+    property double toY: 0
+    property double position: 0
+    property bool handleMargins: false
+    property double rotationDelta: 0  // The actual rotation difference to apply
+    property var completedCallback: null  // Optional callback function to run after jump completes
+
+    onPositionChanged: {
+      if (!enabled) {
+        return;
+      }
+      const dx = toX - fromX;
+      const dy = toY - fromY;
+      const distance = Math.sqrt(dx * dx + dy * dy);
+      if (distance === 0 && toRotation === -1 && toScale === -1) {
+        enabled = false;
+        mapArea.unfreeze('jumping');
+        return;
+      }
+      const progressX = fromX + dx * position;
+      const progressY = fromY + dy * position;
+
+      // Handle rotation animation using the calculated delta
+      if (toRotation !== -1) {
+        const progressRotation = fromRotation + rotationDelta * position;
+        mapCanvasWrapper.mapSettings.rotation = progressRotation;
+      }
+      if (toScale !== -1) {
+        // Handle scale animation
+        const progressScale = fromScale + (toScale - fromScale) * position;
+        mapCanvasWrapper.zoomScale(Qt.point(progressX, progressY), progressScale, jumpDetails.handleMargins);
+      } else {
+        // Handle only position animation
+        mapCanvasWrapper.mapSettings.setCenter(Qt.point(progressX, progressY), jumpDetails.handleMargins);
+      }
+      if (position >= 1.0) {
+        enabled = false;
+        mapArea.unfreeze('jumping');
+        if (completedCallback && typeof completedCallback === 'function') {
+          completedCallback();
+          completedCallback = null;
+        }
+      }
+    }
+
+    Behavior on position  {
+      enabled: jumpDetails.enabled
+      NumberAnimation {
+        easing.type: Easing.InOutQuart
+        duration: 2500
```

Comment:
This feels too long to begin with. I think I'd reduce that to at least 1.5sec if not 1.0sec.

We could tweak the duration to be slightly longer / short based on screen travel distance but let's first get this in.